src/Type/DrupalClassResolverDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,73 @@
+<?php declare(strict_types=1);
+
+namespace mglaman\PHPStanDrupal\Type;
+
+use Drupal\Core\DependencyInjection\ClassResolverInterface;
+use PhpParser\Node\Expr\MethodCall;
+use mglaman\PHPStanDrupal\Drupal\DrupalServiceDefinition;
+use mglaman\PHPStanDrupal\Drupal\ServiceMap;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+
+class DrupalClassResolverDynamicReturnTypeExtension implements DynamicMethodReturnTypeExtension
+{
+    /**
+     * @var ServiceMap
+     */
+    private $serviceMap;
+
+    public function __construct(ServiceMap $serviceMap)
+    {
+        $this->serviceMap = $serviceMap;
+    }
+
+    public function getClass(): string
+    {
+        return ClassResolverInterface::class;
+    }
+
+    public function isMethodSupported(MethodReflection $methodReflection): bool
+    {
+        return $methodReflection->getName() === 'getInstanceFromDefinition';
+    }
+
+    public function getTypeFromMethodCall(
+        MethodReflection $methodReflection,
+        MethodCall $methodCall,
+        Scope $scope
+    ): Type {
+        if (0 === \count($methodCall->args)) {
+            return ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
+        }
```

Comment:
See https://github.com/phpstan/phpstan/releases/tag/1.0.0, we need to call `->getArgs()`

Reply:
Done
Missed that one thanks!